src/js/covid-chart.js

```diff
@@ -0,0 +1,150 @@
+import createElement from './utils/create-element';
+import Switchers from './switchers';
+import FullScreenButton from './full-screen-button';
+import SELECTS from './const/selects';
+import PARAMETERS from './const/parameters';
+import DATASET_INDEXES from './const/dataset-indexes';
+import addCommas from './utils/add-commas';
+import countPer100k from './utils/count-per-100k';
+import Chart from '../../node_modules/chart.js/dist/Chart.bundle';
+import keyboard from './keyboard';
+
+/**
+ * Get markup for the Covid Chart.
+ * @returns {string} String with Covid Chart markup.
+ */
+function getCovidChartMarkup() {
+  const switchers = new Switchers(SELECTS.chartSelects).element;
+  return `${switchers.outerHTML}
+          <div class = "chart__bodyCanvas"><canvas class="covid-chart__canvas" ></canvas></div>`;
+}
+
+/**
+ * Class representing the Covid Chart.
+ */
+export default class CovidChart {
+  constructor() {
+    this.element = createElement('div', 'covid-chart', getCovidChartMarkup());
+    this.canvas = this.element.querySelector('.covid-chart__canvas');
+    this.ctx = this.canvas.getContext('2d');
+    this.currentChart = null;
+    this.switchersContainer = this.element.querySelector('.switchers');
+    this.switchersContainer.append(new FullScreenButton().element);
+    this.fullScreenButton = this.element.querySelector('.full-screen-button');
+    this.bindFullScreen();
+
+    this.country = null;
+    this.valueName = PARAMETERS.valueName.cases;
+    this.period = PARAMETERS.period.allTime;
+    this.valueType = PARAMETERS.valueType.absolute;
+  }
+
+  /**
+   * Draw new Covid Chart.
+   * @param {string} country - A country name. If there is none, a global chart will be drawn.
+   * @param {object} dataPromise - Promise with datasets.
+   * @returns {object} The chart.
+   */
+  buildChart(country, dataPromise) {
+    this.element.classList.toggle('load');
+    dataPromise
+      .then((datasets) => {
+        let historicalData;
+        if (country) {
+          historicalData = datasets[DATASET_INDEXES.historical].timeline;
+        } else {
+          historicalData = datasets[DATASET_INDEXES.historical];
+        }
+        const { population } = datasets[DATASET_INDEXES.today];
+        const dataLabels = Object.keys(historicalData[this.valueName]).map((key) => new Date(key));
+        let dataValues = Object.values(historicalData[this.valueName]);
+        if (this.period === PARAMETERS.period.lastDay) {
+          dataValues = dataValues.map((value, index, arr) => {
+            let result = value - arr[index - 1] || 0;
+            if (result < 0) {
+              result = 0;
+            }
+            return result;
+          });
+        }
+        if (this.valueType === PARAMETERS.valueType.per100k) {
+          dataValues = dataValues.map((value) => countPer100k(value, population));
+        }
+        const chart = new Chart(this.ctx, {
```

Comment:
The good practice is to move a config into a separate file near this class. It useful if the config doesn't change often. Also, your class will be shorter, I recommend to try always to make your functions/classes the shortest as you can.